profiles/serializers.py

```diff
@@ -204,6 +204,15 @@ class Meta:
         )
         read_only_fields = ("person",)
 
+    def validate_for_delete(self, data):
+        password = data.get("password")
+        if not password:
+            raise serializers.ValidationError("Password is required")
+        user = self.context["request"].user
+        if not user.check_password(password):
+            raise serializers.ValidationError("Invalid password")
```

Comment:
This is not exactly correct. There should be a field name, + probably validation error should not be a string, but a list of strings (with one string in the list).

Reply:
This code is outdated. Here is how code looks like now: 
if not user.check_password(password):
            raise serializers.ValidationError({"password": "Invalid password"})

Reply:
Replaced with field validation.